File: scripts/generate-data-js.mjs
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const root = path.resolve(path.dirname(fileURLToPath(import.meta.url)), '..');
const dataDir = path.join(root, 'data');

for (const name of ['config', 'copy', 'emotions', 'community-phrases', 'frasi-sesto-senso']) {
    const json = JSON.parse(fs.readFileSync(path.join(dataDir, `${name}.json`), 'utf8'));
    const js = [
        '(function (w) {',
        '  w.SESTOSENS_DATA = w.SESTOSENS_DATA || {};',
        `  w.SESTOSENS_DATA.${name.replace(/-([a-z])/g, (_, c) => c.toUpperCase())} = ${JSON.stringify(json)};`,
        '})(window);',
        ''
    ].join('\n');
    fs.writeFileSync(path.join(dataDir, `${name}.js`), js);
    console.log(`Wrote data/${name}.js`);
}

File: scripts/generate-data-js.py
#!/usr/bin/env python3
"""Generate data/*.js from data/*.json for file:// compatibility."""

import json
from pathlib import Path

ROOT = Path(__file__).resolve().parent.parent
DATA = ROOT / 'data'

for name in ('config', 'copy', 'emotions'):
    payload = json.loads((DATA / f'{name}.json').read_text(encoding='utf-8'))
    js = '\n'.join([
        '(function (w) {',
        '  w.SESTOSENS_DATA = w.SESTOSENS_DATA || {};',
        f'  w.SESTOSENS_DATA.{name} = {json.dumps(payload, ensure_ascii=False)};',
        '})(window);',
        ''
    ])
    (DATA / f'{name}.js').write_text(js, encoding='utf-8')
    print(f'Wrote data/{name}.js')
